sdk/cosmos/azure-cosmos/azure/cosmos/_timeout_failover_retry_policy.py

```diff
@@ -43,24 +45,61 @@ def ShouldRetry(self, _exception):
         if not self.connection_policy.EnableEndpointDiscovery:
             return False
 
+        self.retry_count += 1
```

Comment:
So, one possible issue with L48 is in multi-write accounts, a write request will only be retried within a region since `_max_service_unavailable_retry_count` is 1 .

Reply:
Great catch, fixed it.